scripts/build-dbt.py

```diff
@@ -0,0 +1,829 @@
+            )
+            self._template = template
+        else:
+            template = self._template
+        return template
+
+    def _get_pypi_dbt_info(self) -> Tuple[str, str]:
+        fp = urlopen(f'https://pypi.org/pypi/dbt/{self.version}/json')
+        try:
+            data = json.load(fp)
+        finally:
+            fp.close()
+        assert 'urls' in data
+        for pkginfo in data['urls']:
+            assert 'packagetype' in pkginfo
+            if pkginfo['packagetype'] == 'sdist':
+                assert 'url' in pkginfo
+                assert 'digests' in pkginfo
+                assert 'sha256' in pkginfo['digests']
+                url = pkginfo['url']
+                digest = pkginfo['digests']['sha256']
+                return url, digest
+        raise ValueError(f'Never got a valid sdist for dbt=={self.version}')
+
+    def _get_recursive_dependencies(self, poet_exe: Path) -> str:
+        cmd = [str(poet_exe), '--resources', 'dbt']
+        raw = collect_output(cmd).split('\n')
+        return '\n'.join(self._remove_dbt_resource(raw))
+
+    def _remove_dbt_resource(self, lines: List[str]) -> Iterator[str]:
+        # TODO: fork poet or extract the good bits to avoid this
+        line_iter = iter(lines)
+        # don't do a double-newline or "brew audit" gets mad
+        for line in line_iter:
+            # skip the contents of the "dbt" resource block.
+            if line.strip() == 'resource "dbt" do':
+                for skip in line_iter:
+                    if skip.strip() == 'end':
+                        # skip the newline after 'end'
+                        next(line_iter)
+                        break
+            yield line
+
+    def create_versioned_formula_file(self):
+        formula_contents = self.get_formula_data(versioned=True)
+        if self.versioned_formula_path.exists():
+            print('Homebrew formula path already exists, overwriting')
+        self.versioned_formula_path.write_text(formula_contents)
+
+    def commit_versioned_formula(self):
+        # add a commit for the new formula
+        run_command(
+            ['git', 'add', self.versioned_formula_path],
+            cwd=self.homebrew_path
+        )
+        run_command(
+            ['git', 'commit', '-m', f'add dbt@{self.version}'],
+            cwd=self.homebrew_path
+        )
+
+    def commit_default_formula(self):
+        run_command(
+            ['git', 'add', self.default_formula_path],
+            cwd=self.homebrew_path
+        )
+        run_command(
+            ['git', 'commit', '-m', f'upgrade dbt to {self.version}'],
+            cwd=self.homebrew_path
+        )
+
+    @staticmethod
+    def run_tests(formula_path: Path):
+        path = os.path.normpath(formula_path)
+        run_command(['brew', 'uninstall', '--force', path])
+        run_command(['brew', 'install', path])
+        run_command(['brew', 'test', path])
+        run_command(['brew', 'audit', '--strict', path])
```

Comment:
idk if you've seen this yet, but be careful running these commands on your macbook. some of these brew commands will completely obliterate everything you have installed via brew

Reply:
Interesting, nothing bad has happened to me yet except postgres getting upgraded on me. I don't use brew for much except postgres. I guess until we get CI doing this part of it, that's just a risk I'll take.